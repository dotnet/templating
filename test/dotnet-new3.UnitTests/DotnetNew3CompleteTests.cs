// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

using FluentAssertions;
using Microsoft.NET.TestFramework.Assertions;
using Microsoft.TemplateEngine.TestHelper;
using Xunit.Abstractions;

namespace Dotnet_new3.IntegrationTests
{
    [UsesVerify]
    [Collection("Verify Tests")]
    public class DotnetNew3CompleteTests
    {
        private readonly ITestOutputHelper _log;

        public DotnetNew3CompleteTests(ITestOutputHelper log)
        {
            _log = log;
        }

        [Fact]
        public Task CanDoTabCompletion()
        {
            string homeDir = TestUtils.CreateTemporaryFolder();
            var commandResult = new DotnetNewCommand(_log, "complete", $"new3 --debug:custom-hive {homeDir} ")
                .WithoutCustomHive()
                .Execute();

            commandResult
                .Should()
                .ExitWith(0)
                .And.NotHaveStdErr();

            return Verify(commandResult.StdOut);
        }

#pragma warning disable xUnit1004 // Test methods should not be skipped
        [Fact(Skip = "https://github.com/dotnet/command-line-api/issues/1519")]
#pragma warning restore xUnit1004 // Test methods should not be skipped
        public void CanDoTabCompletionAtGivenPosition()
        {
            string homeDir = TestUtils.CreateTemporaryFolder();
            var commandResult = new DotnetNewCommand(_log, "complete", $"new3 co --debug:custom-hive {homeDir} --language C#", "--position", "7")
                .WithoutCustomHive()
                .Execute();

            commandResult
                .Should()
                .ExitWith(0)
                .And.NotHaveStdErr()
                .And.HaveStdOut("console");
        }
    }
}
