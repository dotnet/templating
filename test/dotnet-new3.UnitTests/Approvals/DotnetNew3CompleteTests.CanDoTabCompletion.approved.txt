classlib
console
editorconfig
gitignore
globaljson
nugetconfig
sln
tool-manifest
webconfig
install
list
search
uninstall
update
--help
-?
-h
/?
/h